### DISCUSSION TITLE
Constantly high CPU usage on Intel Mac (hypervisor)

### DISCUSSION THREAD

**Discussion by tiiiecherle** [utmapp/UTM#3611] [UTM Help (macOS)]
Hey,

first of all thanks a lot for this project, really appreciated.

As I am going to buy a new MacBook Pro with M1 I was switching from Virtualbox to UTM on my current MacBook Pro 2019, 1 TB SSD, 32 GB Ram.

I am not an expert but my vms are x86_64 and I am on an Intel Mac. Is there a way to use UTM as a hypervisor and not an emulator in this setup? 

The performance in the Windows 10 vm and the Archlinux vm is very (s)low and uses a way more cpu than Virtualbox. Even if the vm is using not so much cpu in the activity monitor it is constantly draining the battery pretty fast and even when not using the vm at all the fans do not stop spinning. There are no active processes in the vm besides the base system running.

When trying out settings I found out that scaling set to nearest neighbour gives me way better and sharper graphics. I don`t know about other systems, but regarding to quality here it should be the default value.

I am using the following settings:
- standard q35 pc
- 8.196 MB RAM
- cpu default
- no separately enabled cpu flags
- drives as disk image in qcow2 format
- full graphics with virtio-vga-gl card
- Scaling both nearest neighbour
- sound enabled
- clipboard sharing, directory sharing and usb3 enabled

Thanks in advance for any feedback and help.

**Comment by tiiiecherle** [comment]
Additional Information:
UTM Version 2.4.1
macOS Version 11.6.1
uploaded config.plist of Windows 10 vm

**Comment by tiiiecherle** [comment]
<details>
  <summary>VM Config PLIST</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>ConfigurationVersion</key>
	<integer>2</integer>
	<key>Debug</key>
	<dict/>
	<key>Display</key>
	<dict>
		<key>ConsoleFont</key>
		<string>Menlo</string>
		<key>ConsoleFontSize</key>
		<integer>12</integer>
		<key>ConsoleOnly</key>
		<false/>
		<key>ConsoleTheme</key>
		<string>Default</string>
		<key>DisplayCard</key>
		<string>virtio-vga-gl</string>
		<key>DisplayDownscaler</key>
		<string>nearest</string>
		<key>DisplayRetina</key>
		<false/>
		<key>DisplayUpscaler</key>
		<string>nearest</string>
	</dict>
	<key>Drives</key>
	<array>
		<dict>
			<key>DriveName</key>
			<string>drive0</string>
			<key>ImagePath</key>
			<string>win10_64.qcow2</string>
			<key>ImageType</key>
			<string>disk</string>
			<key>InterfaceType</key>
			<string>ide</string>
		</dict>
		<dict>
			<key>DriveName</key>
			<string>drive1</string>
			<key>ImageType</key>
			<string>cd</string>
			<key>InterfaceType</key>
			<string>ide</string>
			<key>Removable</key>
			<true/>
		</dict>
	</array>
	<key>Info</key>
	<dict>
		<key>Icon</key>
		<string>windows</string>
		<key>IconCustom</key>
		<false/>
	</dict>
	<key>Input</key>
	<dict>
		<key>InputLegacy</key>
		<false/>
	</dict>
	<key>Networking</key>
	<dict>
		<key>NetworkBridgeInterface</key>
		<string>en0</string>
		<key>NetworkCard</key>
		<string>e1000</string>
		<key>NetworkCardMAC</key>
		<string>3E:C6:4F:34:68:B8</string>
		<key>NetworkMode</key>
		<string>bridged</string>
	</dict>
	<key>Printing</key>
	<dict/>
	<key>Sharing</key>
	<dict>
		<key>ClipboardSharing</key>
		<true/>
		<key>DirectorySharing</key>
		<true/>
		<key>Usb3Support</key>
		<true/>
		<key>UsbRedirectMax</key>
		<integer>3</integer>
	</dict>
	<key>Sound</key>
	<dict>
		<key>SoundCard</key>
		<string>intel-hda</string>
		<key>SoundEnabled</key>
		<true/>
	</dict>
	<key>System</key>
	<dict>
		<key>Architecture</key>
		<string>x86_64</string>
		<key>BootDevice</key>
		<string></string>
		<key>BootUefi</key>
		<false/>
		<key>CPU</key>
		<string>default</string>
		<key>JITCacheSize</key>
		<integer>0</integer>
		<key>MachineProperties</key>
		<string>vmport=off</string>
		<key>Memory</key>
		<integer>8196</integer>
		<key>SystemUUID</key>
		<string>796AD07D-8E7C-4419-8929-7F39548B5325</string>
		<key>Target</key>
		<string>q35</string>
	</dict>
</dict>
</plist>

```

**Comment by conath** [comment]
Welcome to UTM.

> I am not an expert but my vms are x86_64 and I am on an Intel Mac. Is there a way to use UTM as a hypervisor and not an emulator in this setup?

Yes, if your host and guest architecture is the same then UTM will by default use the macOS Hypervisor instead of emulating the CPU completely. This means very fast performance. The same will NOT work if your host is not the same as guest, e. g. Apple Silicon CPU and a x86 VM.

Now, in terms of CPU usage with hypervisor mode, on Intel Mac I have experienced the same (unusually high CPU usage even if the VM is idling). On Apple Silicon this is not the case, so maybe a problem with the Apple Hypervisor or QEMU (UTM backend).

**Comment by conath** [comment]
@tiiiecherle can you try again with CPU set to "host"? And please provide Debug Log as per [VM Troubleshooting](https://github.com/utmapp/UTM/wiki/Troubleshooting-a-VM#how-to-get-debug-log) guide.

**Comment by tiiiecherle** [comment]
Thanks for the help and the suggestions. I tried running the vms with CPU set to host and did not get a really noticeable difference except the win 10 vm crashes a lot.
If it helps I can try to get a debug log in the next days.
But the performance/speed is still a lot less than on virtualbox and opening an app on win takes quite a while and if anything else is used it even takes a lot longer.

**Comment by svensen** [comment]
It seems qemu in UTM is runnning with  `-accel hvf -accel tcg,tb-size=2048` when runnning x86_64 guests on x86_64 hosts. 
Is "accel=tcg" the right way when running on native CPU?

**Reply by osy** [reply]
It’s fine as hvf is first. But I think high cpu usage is normal.

**Reply by arroz** [reply]
Hi Osy,

I'm seeing the same issue running guest FreeBSD with a M1 host. Some data:

- This doesn't seem to happen when running QEMU directly using -accel hvf, it only happens when running via UTM.
- It's hard to tell if this is just a bogus usage based on the guest/host idle thread issue, but if I run a VM using 4 cores, all the Performance cores of my M1 move from idle frequency (~1.8 Ghz) to a higher one (2.7-3.2 Ghz), so I assume they are indeed doing something and wasting power.

**Reply by jamestut** [reply]
Hi @arroz, yes it is indeed wasting power and not "bogus" usage. It is very apparent when monitoring the CPU power usage via `powermetrics` (can use this command to quickly view CPU power usage in terminal `sudo powermetrics -s cpu_power -i 2000 | fgrep -i power`).

**Reply by arroz** [reply]
Ah, I just found out the problem in my use case (well, somebody pointed it out to me because apparently I can't read the manual… 🤦‍♂️). When using FreeBSD as the guest OS, one needs to add to the `/boot/loader.conf` file the following line: `kern.hz=100`. This is documented in https://docs.freebsd.org/en/books/handbook/virtualization/ and drops the CPU usage while idle from 300-400% to nearly 0 (around 4-5%). There's still some overhead which maybe coming from the GUI console, but it's much better than blasting half of the cores to their limit. 🙂

**Reply by numbermaniac** [reply]
I'm having a similar problem when trying to run either Ubuntu or Debian as the guest OS. Especially when UTM is in the background (not the foreground window), at some point the VM will just randomly freeze and UTM will spike to 100% CPU usage and it'll stay there forever. There's nothing you can do to recover the situation, other than forcefully shut down the VM and lose anything that was in progress. UTM is proving to be rather difficult to use over QEMU, despite supposedly being the "easy way" to use QEMU.
